pkg/kp/kv.go

```diff
@@ -0,0 +1,132 @@
+// Package kp provides a generalized API for reading and writing pod manifests
+// in consul.
+package kp
+
+import (
+	"bytes"
+	"time"
+
+	"github.com/armon/consul-api"
+	"github.com/square/p2/pkg/pods"
+	"github.com/square/p2/pkg/util"
+)
+
+type ManifestResult struct {
+	Manifest pods.PodManifest
+	Path     string
+}
+type Options struct {
+	Token   string
+	Address string
+}
+
+type Store struct {
+	client *consulapi.Client
+}
+
+func NewStore(opts Options) *Store {
+	conf := consulapi.DefaultConfig()
+	conf.Address = opts.Address
+	conf.Token = opts.Token
+
+	// the error is always nil
```

Comment:
I wonder if one day consul will change the behavior of NewClient such that error could be nil. We would be a bit unprepared in that case